wgsl/index.bs

```diff
@@ -546,11 +546,16 @@ with the following exceptions
 
  * No Acquire/Release semantics
 
-### Memory Locations TODO ### {#memory-locations}
-
-TODO: *This is a stub*
-
-Memory consists of distinct locations.
+### Memory Locations ### {#memory-locations}
+
+Memory consists of a set of distinct locations. Each memory location is 8-bits
+in size. An operation affecting memory interacts with a set of one or more
+memory locations. Two sets of memory locations overlap if the intersection of
+their sets of memory locations is non-empty. Each variable declaration has a
+set memory locations that do not overlap with the sets of memory locations of
```

Comment:
Nit: "has a set __of__ memory locations"

More nitty: I think it should be "__does__ not overlap". We're talking about whether a set (singular) of memory locations overlaps with other sets, not whether the memory locations themselves (plural) overlap with other sets.